lib/winapi/profiling.c

```diff
@@ -1,24 +1,95 @@
 // SPDX-License-Identifier: MIT
 
-// SPDX-FileCopyrightText: 2019 Stefan Schmidt
+// SPDX-FileCopyrightText: 2019-2025 Stefan Schmidt
 
 #include <assert.h>
+#include <intrin.h>
 #include <profileapi.h>
 #include <stddef.h>
+#include <stdint.h>
 #include <xboxkrnl/xboxkrnl.h>
 
+static LARGE_INTEGER tsc_freq = {.QuadPart = 733333333};
+
 BOOL QueryPerformanceCounter (LARGE_INTEGER *lpPerformanceCount)
 {
     assert(lpPerformanceCount != NULL);
 
-    lpPerformanceCount->QuadPart = KeQueryPerformanceCounter();
+    lpPerformanceCount->QuadPart = __rdtsc();
     return TRUE;
 }
 
 BOOL QueryPerformanceFrequency (LARGE_INTEGER *lpFrequency)
 {
     assert(lpFrequency != NULL);
 
-    lpFrequency->QuadPart = KeQueryPerformanceFrequency();
+    *lpFrequency = tsc_freq;
     return TRUE;
 }
+
+static uint64_t get_cpu_multiplier (void)
+{
+    // Read EBL_CR_POWERON
+    const uint64_t poweron_reg = __readmsr(0x2A);
+
+    // Chapter 2.22 of https://www.intel.com/content/dam/develop/external/us/en/documents/335592-sdm-vol-4.pdf
+    const uint64_t multiplier_pattern = (poweron_reg >> 22) & 0b101111;
+
+    // https://web.archive.org/web/20040407043044/http://www.intel.com:80/design/mobile/datashts/29834006.pdf
```

Comment:
![image](https://github.com/user-attachments/assets/4ce9cbc2-b428-45e9-905d-4f482ea128b0)

Neither document mentioned in this PR has information on >10X multipliers, this is something I brought up (and wasn't answered) in xemu-project/xemu/pull/1551, I also attempted to search Linux and they too seem not to mention support for >10X. Where are you getting these numbers?

Reply:
From the CPUs, specifically SL6BY, SL6C6 and SL6C7.